Add Travis CI builds for PRs?

IMHO adding [Travis CI](https://travis-ci.org/) support to this project would be helpful - it would help reveal issues like #187 immediately, rather than having to track down their cause later.

@broady @markmcd Thoughts?

If you're willing to turn on Travis support for this project in the Travis CI console, I'd be willing to try configure the `.travis.yml` script.  I've done this on many Android apps, but never on a library before, and its always good to learn new things :).

Here's the latest branch I'm working on:
https://github.com/barbeau/android-maps-utils/tree/travis3

This has the Google API emulator starting and the tests executing - see [this Travis build](https://travis-ci.org/barbeau/android-maps-utils/builds/83048457).  However, it fails with:

~~~
:library:connectedDebugAndroidTest FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':library:connectedDebugAndroidTest'.
> com.android.builder.testing.api.TestException: com.android.builder.testing.api.DeviceException: com.android.ddmlib.ShellCommandUnresponsiveException
~~~

Usually this means ADB timed out:
http://stackoverflow.com/a/28949723/937715

The fix for this is normally to increase ADB timeout via the Travis variable `ADB_INSTALL_TIMEOUT `.

However, I've bumped it up to 15 min so far and I'm still getting the same exception on Travis.

Also, not sure if its related to these issues, but I did find [this open AOSP issue](https://code.google.com/p/android/issues/detail?id=176348&thanks=176348&ts=1433887196) saying that Google Play Services is broken on the Google API emulator starting with API Level 22.  It's been triaged but is still open, so the same issue may exist on API Level 23 emulator.

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,47 @@
+language: android
+jdk: oraclejdk7
+# Use the Travis Container-Based Infrastructure
+sudo: false
+
+cache:
+  directories:
+    - ${TRAVIS_BUILD_DIR}/gradle/caches/
+    - ${TRAVIS_BUILD_DIR}/gradle/wrapper/dists/
+
+env:
+  global:
+    - ANDROID_API_LEVEL=23
+    - EMULATOR_API_LEVEL=21
+    - ANDROID_BUILD_TOOLS_VERSION=23.0.0
+    - ANDROID_ABI=google_apis/armeabi-v7a
+    - ADB_INSTALL_TIMEOUT=20 # minutes (2 minutes by default)
+
+android:
+  components:
+    - platform-tools
+    - tools
+    - build-tools-$ANDROID_BUILD_TOOLS_VERSION
+    - android-$ANDROID_API_LEVEL
+    - android-$EMULATOR_API_LEVEL
+    # For Google APIs
+    - addon-google_apis-google-$ANDROID_API_LEVEL
+    - addon-google_apis-google-$EMULATOR_API_LEVEL
+    # Google Play Services
+    - extra-google-google_play_services
+    # Support library
+    - extra-android-support
+    # Latest artifacts in local repository
+    - extra-google-m2repository
+    - extra-android-m2repository
+    # Specify at least one system image
+    - sys-img-armeabi-v7a-addon-google_apis-google-$ANDROID_API_LEVEL
+    - sys-img-armeabi-v7a-addon-google_apis-google-$EMULATOR_API_LEVEL
+
+before_script:
+  # Create and start emulator
+  - echo no | android create avd --force -n test -t "Google Inc.:Google APIs:"$EMULATOR_API_LEVEL --abi $ANDROID_ABI
+  - emulator -avd test -no-skin -no-audio -no-window &
+  - android-wait-for-emulator
+
+script:
+  - ./gradlew connectedCheck -x library:signArchives -PdisablePreDex

--- a/README.md
+++ b/README.md
@@ -1,4 +1,4 @@
-[![Maven Central](https://maven-badges.herokuapp.com/maven-central/com.google.maps.android/android-maps-utils/badge.svg)](https://maven-badges.herokuapp.com/maven-central/com.google.maps.android/android-maps-utils)
+[![Maven Central](https://maven-badges.herokuapp.com/maven-central/com.google.maps.android/android-maps-utils/badge.svg)](https://maven-badges.herokuapp.com/maven-central/com.google.maps.android/android-maps-utils)[![Build Status](https://travis-ci.org/googlemaps/android-maps-utils.svg?branch=master)](https://travis-ci.org/googlemaps/android-maps-utils)
 
 # Google Maps Android API utility library
 

--- a/build.gradle
+++ b/build.gradle
@@ -7,3 +7,18 @@ buildscript {
         classpath 'com.android.tools.build:gradle:1.3.0'
     }
 }
+/**
+ * Improve build server performance by allowing disabling of pre-dexing
+ * (see http://tools.android.com/tech-docs/new-build-system/tips#TOC-Improving-Build-Server-performance.)
+ */
+project.ext.preDexLibs = !project.hasProperty('disablePreDex')
+
+subprojects {
+    project.plugins.whenPluginAdded { plugin ->
+        if ("com.android.build.gradle.AppPlugin".equals(plugin.class.name)) {
+            project.android.dexOptions.preDexLibraries = rootProject.ext.preDexLibs
+        } else if ("com.android.build.gradle.LibraryPlugin".equals(plugin.class.name)) {
+            project.android.dexOptions.preDexLibraries = rootProject.ext.preDexLibs
+        }
+    }
+}

